common/formats/src/format_csv.rs

```diff
@@ -338,6 +339,16 @@ impl InputFormat for CsvInputFormat {
         Ok(index)
     }
 
+    fn set_buf(&self, buf: Vec<u8>, state: &mut Box<dyn InputState>) {
```

Comment:
It seems more like an API for `InputState`.

Different input states share the same logic.